docs/overview.md

```diff
@@ -0,0 +1,135 @@
+# Introduction
+
+::: tip Note
+**Sprocket** is current an alpha-phase project. To that effect,
+this page serves the purpose of describing what we hope Sprocket _will_ become
+rather than what it actually is today. If you're using Sprocket, we encourage
+you to follow the project on
+[GitHub](https://github.com/stjude-rust-labs/sprocket) to stay up to date on
+progress.
+:::
+
+**Sprocket** is an bioinformatics workflow orchestration engine and package
+manager built on top of the [Workflow Description Language](https://openwdl.org). The project has multiple high-level goals, including to:
+
+* Provide a **high-performance** workflow execution engine capable of
+  orchestrating massive bioinformatics workloads (the stated target is 20,000+
+  concurrent jobs).
+* Remain **simple and accessible user experience** when complexity isn't warranted.
+* Develop a suite of **modern development tools** that brings bioinformatics
+  development on par with other modern languages (e.g.,
+  [`wdl-lsp`](https://github.com/stjude-rust-labs/wdl/tree/main/wdl-lsp)).
+* Maintain an **community-focused codebase** that enables a diverse set of
+  contributors from academic, non-profit, and commercial organizations.
+* Build on an **open, domain-tailored standard** to ensure the toolset remains
+  singularly focused on unencumbered innovation within bioinformatics.
+
+Sprocket is written in [Rust](https://www.rust-lang.org/) and enjoys all of the
+benefits that come with that choice. It also takes heavy inspiration from Rust
+in terms of its approach to building developer tools that are a joy to use. The
+code that drives Sprocket is split across the [`wdl`] family of crates, the
+[`sprocket`] command line tool, and the [Visual Studio Code extension]
+([source](https://github.com/stjude-rust-labs/sprocket-vscode)).
+
+## Project Goals
+
+### High-Performance Orchestration Engine
+
+Most fundamentally, the Sprocket project was created due to the lack of workflow
+engines that were powerful enough to orchestrate tens of thousands or hundreds
+of thousands of concurrent jobs. Enabling the next generation of large-scale,
+open science will always remain the top-level objective for the project.
```

Comment:
I'd prefer that we give this some more thought before putting it out there. What exactly does this look like? There are several execution engines (Cromwell, Miniwdl, dx-compiler, AWS HealthOmics). These have a variety of different target environments and I think it'd be best to define what this looks like and what we want to support.

Reply:
This was discussed extensively in Slack and I've updated the text here to
reflect that discussion.